rest-client/src/main/java/org/sagebionetworks/bridge/rest/ErrorResponseInterceptor.java

```diff
@@ -72,41 +74,46 @@ private void throwExceptionOnErrorStatus(String url, int statusCode, JsonElement
                 message = "There has been an error on the server";
             }
         }
-
-        BridgeSDKException e;
-        if (statusCode == 401) {
-            e = new NotAuthenticatedException(message, url);
-        } else if (statusCode == 403) {
-            e = new UnauthorizedException(message, url);
-        } else if (statusCode == 404) {
-            e = new EntityNotFoundException(message, url);
-        } else if (statusCode == 410) {
-            e = new UnsupportedVersionException(message, url);
-        } else if (statusCode == 412) {
```

Comment:
Are we sure we want to change the error handling code? While there are a lot of error cases where we know for sure are coming from Bridge code, there may be cases where it's not safe to assume Bridge server is properly responding with JSON.

The most recent example is 400s used to return HTML. (This was recently changed with your changelist.) Other examples could include 500s and 503s. It looks like in your code, if we don't return a proper JSON exception, we return a BridgeSDKException (corresponding to an HTTP 500). I think it's worth making sure we have code that even if the server somehow returns a 400 with a non-JSON body, we report it to downstream consumers as a 400 as well.